python/ray/air/checkpoint.py

```diff
@@ -454,6 +471,34 @@ def from_directory(cls, path: str) -> "Checkpoint":
 
         return checkpoint
 
+    @classmethod
+    def from_directory_populator(cls, populator: Callable[[str], Any]):
+        """Create a checkpoint from a populator function.
+
+        The populator function receives one argument, a string pointing to a directory.
+        The function can proceed to write data into this directory.
+
+        This method will then return a Checkpoint created from this directory.
+
+        Args:
+            populator: Function taking a directory path as an input. The populator
+                function can write checkpoint data into this directory.
+
+        Returns:
+            Checkpoint object containing data written to the directory.
+
+        Examples:
+            >>> checkpoint = Checkpoint.from_directory_populator(  # doctest: +SKIP
```

Comment:
How should I call this API if I want to save both a directory and dict checkpoint?

Reply:
say I have both files and dicts to be saved in this checkpoint?

Reply:
In that case, you should pickle your dicts into a file in the directory